src/main/java/com/contrastsecurity/sdk/scan/CodeArtifactInner.java

```diff
@@ -0,0 +1,52 @@
+package com.contrastsecurity.sdk.scan;
+
+import com.google.auto.value.AutoValue;
+import java.time.Instant;
+
+/** Value type that describes the code artifact structure returned by the Contrast API. */
+@AutoValue
+abstract class CodeArtifactInner {
```

Comment:
Opinion question, for these `Inner` classes. What do you think about using an `Api` prefix instead, `ApiCodeArtifact`?

Reply:
I'm not in love with the `Inner` suffix either, but it is [consistent with the types used by the Azure Java SDK](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/fluent/models), which you'll recall is the example we're referencing for this design. Like we say in the Java Team style guide, we like to avoid bike-shedding opinions by adopting some existing standard, so I'm cool with just doing what they do, even if I don't love it.

I do have opinions on suffix vs prefix. I strongly prefer a suffix in this case, because in an alphabetical listing of classes, we want all the classes related to a resource to be grouped together e.g.

```
Project.java
ProjectClient.java
ProjectClientImpl.java
ProjectCreate.java
ProjectImpl.java
ProjectInner.java
Projects.java
ProjectsImpl.java
```

If you use a prefix, then you'll group classes together by their layer archetype instead of by resource. In this resource-centric design, we want to package by feature and group by resource.

Reply:
Bryan's naming convention is one I do like more as far as illustrating what the object really is, I do see the point regarding suffix vs prefix naming though. While `Api` is more descriptive imo it is unfortunately not fit for a suffix in a concise way. I can also see us going down a naming rabbit hole that is similar to what we currently have. We may as well just follow the pattern we are attempting to emulate for the time being

Reply:
I'm fine with following the established pattern